tensorflow_lite_support/ios/task/audio/sources/TFLAudioClassifier.h

```diff
@@ -0,0 +1,175 @@
+ * specific requirement.
+ *
+ * @param modelPath Path to the model file.
+ *
+ * @param error An optional error parameter populated if there is an error in initializing the audio
+ * classifier.
+ *
+ * @return A `TFLAudioClassifier` instance if initialization is successful or nil in case of
+ * failure.
+ */
+- (nullable instancetype)initWithModelPath:(NSString *)modelPath error:(NSError **)error;
+
+/**
+ * Initializes a new `TFLAudioClassifier` with the given `TFLAudioClassifierOptions`.
+ *
+ * @param options `TFLAudioClassifierOptions` initized with the absolute path to a model file stored
+ * locally on the device and other custom options for configuring the audio classifier.
+ *
+ * @param error An optional error parameter populated when there is an error in initializing the
+ * audio classifier.
+ *
+ * @return A `TFLAudioClassifier` instance if initialization is successful or nil in case of
+ * failure.
+ */
+- (nullable instancetype)initWithOptions:(TFLAudioClassifierOptions *)options
+                                   error:(NSError *_Nullable *)error;
+
+/**
+ * Creates a `TFLAudioTensor` instance to store the input audio samples to be classified. The
+ * created `TFLAudioTensor` has the same buffer size as the model input tensor and audio format
+ * required by the model.
+ *
+ * @param error An optional error parameter populated when there is an error in creating the audio
+ * tensor.
+ *
+ * @return A `TFLAudioTensor` with the same buffer size as the model input tensor and audio format
+ * required by the model, if creation is successful otherwise nil.
+ */
+- (nullable TFLAudioTensor *)createInputAudioTensorWithError:(NSError **)error;
+
+/**
+ * Creates a `TFLAudioRecord` instance to start recording audio input from the microphone. The
+ * returned `TFLAudioRecord` instance instance is initialized with the audio format and twice the
+ * input buffer size required by the model.
+ *
+ * @discussion After creating the `TFLAudioRecord` using this function, the client needs to call
+ * -[TFLAudioRecord startRecordingWithError:] to start recording the audio from the microphone.
+ *
+ * @param error An optional error parameter populated if there is an error in creating the audio
+ * record.
+ *
+ * @return A `TFLAudioRecord` instance with audio format and twice the input buffer size required by
+ * the model, if creation is successful or nil in case of failure.
+ */
+- (nullable TFLAudioRecord *)createAudioRecordWithError:(NSError **)error;
+
+/**
+ * Returns the required input buffer size in number elements or zero in case of failure to determine
+ * the input buffer size.
+ *
+ * @param error An optional error parameter populated if there is an error in determining the input
+ * buffer size.
+ *
+ * @return Required input buffer size in number elements or zero in case of failure to determine the
+ * input buffer size.
+ */
+- (NSInteger)requiredBufferSizeWithError:(NSError **)error;
```

Comment:
As we already provided createInputAudioTensorWithError method to create the appropriate input audio tensor, we shouldn't expose requiredBufferSizeWithError and requiredTensorFormatWithError to avoid making the API unnecessarily complex.